tests/test_human_player.py

```diff
@@ -1,49 +1,44 @@
+import pytest
 from ttt.players.human_player import HumanPlayer
 from ttt.console_ui.console import Console
 from tests.mocks import MockConsole, MockGame, MockConsoleIO
 
 
-def test_human_player_moves_get_an_integer_from_the_console():
-    console = MockConsole()
-    player = HumanPlayer('player 1', 'X', console)
+class TestRunner:
+    def set_player_marker(self, inputs):
+        io = MockConsoleIO(inputs)
+        console = Console(io)
+        player = HumanPlayer('Player 1', 'X', console)
+        player.set_marker()
 
-    player.get_move(MockGame(available_spaces=[]))
+        return player.get_marker()
 
-    assert console.get_valid_input_call_count == 1
 
+@pytest.fixture
+def runner():
+    return TestRunner()
 
-def test_a_player_can_select_a_custom_marker():
-    io = MockConsoleIO(['!'])
-    console = Console(io)
-    player = HumanPlayer('Player 1', 'X', console);
-    player.set_marker()
 
-    assert player.get_marker() == '!'
+def test_human_player_moves_get_an_integer_from_the_console():
+    console = MockConsole()
+    player = HumanPlayer('player 1', 'X', console)
 
+    player.get_move(MockGame(available_spaces=[]))
 
-def test_a_player_will_continue_to_be_prompted_if_they_provide_an_integer_as_a_marker():
-    io = MockConsoleIO(['1', '3', '5', '!'])
-    console = Console(io)
-    player = HumanPlayer('Player 1', 'X', console);
-    player.set_marker()
+    assert console.get_valid_input_call_count == 1
 
-    assert player.get_marker() == '!'
 
+def test_a_player_can_select_a_custom_marker(runner):
+    assert runner.set_player_marker(['!']) == '!'
 
-def test_a_player_will_continue_to_be_prompted_if_they_provide_whitespace_as_a_marker():
-    io = MockConsoleIO([' ', ' ', '   ', '!'])
-    console = Console(io)
-    player = HumanPlayer('Player 1', 'X', console);
-    player.set_marker()
 
-    assert player.get_marker() == '!'
+def test_a_player_will_continue_to_be_prompted_if_they_provide_an_integer_as_a_marker(runner):
+    assert runner.set_player_marker(['1', '3', '5', '!']) == '!'
```

Comment:
I find this quite misleading because the test makes it appear that the runner is responsible for setting the players marker.

How could you re-write these tests so that you provide a sequence of the inputs you want to test?